include/arch/arch_inlines.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * arch_inlines.h - automatically selects the correct arch_inlines.h file to
+ *   include based on the selected architecture.
+ */
+
+/*
+ * Copyright (c) 2019 Stephanos Ioannidis <root@stephanos.io>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_INLINES_H_
```

Comment:
why do we need this header?
can't we just put in some header derived from arch/cpu.h?

Reply:
The rationale for not including `arch/arch_inlines.h` from a header derived from `arch/cpu.h` is as follows:
1. The purpose of `arch/arch_inlines.h` is to provide implementations for the architecture-specific inline functions that are part of the public arch interface defined by `sys/arch_interface.h`.
2. Since `sys/arch_interface.h` only provides prototype and documentation for the inline functions, and all inline functions must have a body defined prior to use, it only makes sense to include `arch/arch_inlines.h` from `sys/arch_interface.h` to ensure that the inline function bodies are defined prior to use.
3. Since `sys/arch_interface.h` includes `arch/cpu.h` at the top, if we include `arch_inlines.h` from a header derived from `arch/cpu.h`, because there could be some functions inside the `arch_inlines.h` that require the function prototypes defined in `sys/arch_interface.h`, we have to either manually do forward declaration of such functions (ugly!), or include `sys/arch_interface.h` from the `arch_inlines.h` (circular dependency!).

For the above reasons, I made sure to keep `arch_inlines.h` as independent as possible. In fact, `sys/arch_interface.h` is the only header that should include `arch_inlines.h` (at the bottom, after all arch interface function prototypes are declared), as the function implementations defined by `arch_inlines.h` are part of the public arch interface, and inline functions always require a body to work.